### PR TITLE
[Android] Show keyboard when Entry, Editor and SearchBar get focus

### LINKED ISSUE #5983: Soft Keyboard does not Pop Up when Entry View's Focus is set to True Programmatically
### Description

When Focus is set to an Entry view programmatically (entryTest.Focus()), the cursor blinks in the edit field of the Entry view, but the soft keyboard does not pop up as it should until you physically touch the Entry view.  In comparison to Xamarin Forms where the keyboard pops up when executing entryTest.Focus().

Also, entryTest.Focus() is ignored when placed in the OnAppearing Event.

Video demo of Maui App:

https://user-images.githubusercontent.com/101732760/162827580-c62e0530-9fa1-4f90-b744-90c37caeec57.mp4

Video demo of Xamarin Forms version:

https://user-images.githubusercontent.com/101732760/162827635-9904fcb9-7276-4eba-8d90-ac4c2c7019d9.mp4

XAML:
image

Code Behind:
image

### Steps to Reproduce

1.  Create a new blank Maui app
2.  Add and Entry View and name it entryTest
3.  Add 2 Buttons and set Text to "Set Focus" and "Set Unfocus"
4.  In the Code behind for the 1st button add code entryTest.Focus()
5. In the Code behind for the 2nd button add code enryTest.Unfocus()
6.  Run the App
7.  Click the Set Focus button.
8.  Notice the Entry field cursor blinks
9.  Notice the keyboard does not pop up
10.  Tap the Entry field.
11.  Notice the keyboard pops up.

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

### Did you find any workaround?

Workaround

I added a mapper [here](https://github.com/PureWeen/ShanedlerSamples/blob/main/ShanedlerSamples/Library/Workarounds/EntryWorkarounds.Android.cs) that you can copy to add this behavior for android. 

One thing to note is that if your intention is to open the keyboard you should use
KeyboardManager.ShowKeyboard not `Focus`

Using `Focus` to influence the `Keyboard` has always felt like a hack to me in general. 

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

 Show/hide keyboard when Entry, Editor and SearchBar get or lost focus.

issue-5983

### Issues Fixed

Fixes #5983

### CODE REVIEW

File: src/Core/src/Handlers/Editor/EditorHandler.Android.cs [hunk 1]
```diff
@@ -95,6 +95,23 @@ void OnPlatformViewAttachedToWindow(object? sender, ViewAttachedToWindowEventArg
...
+			if (e.HasFocus)
+			{
+				// Post this to the main looper queue so it doesn't happen until the other focus stuff has resolved
+				// Otherwise, ShowKeyboard will be called before this control is truly focused, and we will potentially
+				// be displaying the wrong keyboard
+				platformView?.PostShowKeyboard();
+			}
+			else
+			{
+				platformView?.HideKeyboard();
```
Comment (PureWeen) [inline src/Core/src/Handlers/Editor/EditorHandler.Android.cs:113]: What happens here if you are tabbing through multiple fields?

**Review by PureWeen — CHANGES_REQUESTED** [review]
Keyboard visibility was very broken in XF. I don't think we should just bring this code straight over. Someone also logged this same bug on XF against iOS https://github.com/xamarin/Xamarin.Forms/issues/8952

File: src/Core/src/Handlers/Editor/EditorHandler.Android.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Core/src/Handlers/Editor/EditorHandler.Android.cs:113]: If we have something like:

StackLayout
- Entry
- Entry
- Entry

If we have the focus on the first entry, and pass the focus to the second one:
- The first Entry loses focus and the keyboard closes.
- The second Entry gains focus so the keyboard is reopened.

That is, the keyboard would close and open in a short time.

Comment (PureWeen) [reply src/Core/src/Handlers/Editor/EditorHandler.Android.cs:113]: I don't feel like that's the behavior we'd want here. If we want to maintain some level of behavior here with MAUI then I would say at the controls level if the user calls "FOCUS" then we open the keyboard but then that's it.

**Comment by jsuarezruiz** [comment]
To help focus and move faster, we're going to have fewer open PRs and focus on the ones we'll be working on in each release. This PR is not included in the next release so I close it for now. I will reopen it and adapt to changes later.

**Pull request closed without merging**